### ISSUE TITLE
Incorrect Comment Syntax for `<script>` Tag

### ISSUE THREAD

**Issue by ryanfitzer** [textmate/html.tmbundle#48]
**Steps to reproduce**:

  1. Create a new document.
  2. Change language to HTML.
  3. Add an HTML skeleton to the page with a `<script>` tag inside the `<body>` tag:

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta http-equiv="Content-type" content="text/html; charset=utf-8">
  <title>Example</title>
</head>
<body>
  <script src="main.js"></script>
</body>
</html>
```

  4. Place your cursor in front of `<script>` tag and invoke the comment command by pressing `⌘?`.

**Expected result**:
  
The line is commented with the correct HTML syntax: `<!-- <script src="main.js"></script> -->`

**Actual result**:

The line is instead commented with the incorrect syntax: `// <script src="main.js"></script>`.

The issue is also reproducible when:
  
  - The `<script>` tag is inside the `<head>` tag.
  - The `<script>` tag is selected.
  - The full line containing `<script>` tag is selected.

The expected result is only possible when the selection starts on the previous line and includes the full `<script>` tag.

**Environment**:  

  - OS X version: Yosemite 10.10.5
  - TextMate version: 2.0-rc.1

**Comment by aeschli** [comment]
It would be great if the grammar contains a special scope just for the script element content.

**Comment by sorbits** [comment]
The comment command read comment characters via scoped environment variables.

By default, scope selectors match the scope to the right of the caret, which here is the `source.js.embedded`.

If we change the scope selector for _JavaScript → Settings → Comments_ to `L:source.js` then the JavaScript comment characters will not be active when caret is to the left of a `<script>` tag. However, the default `source` characters will then be used, so the scope selector of _Source → Settings → Comments_ must also be changed to have the `L:` prefix.

Once this is done, you can use ⌘/ to toggle comments with caret at the start of a `<script>` tag, and it will wrap it in `<!-- … -->`.

Hint: To “debug” this, place caret at the start of the `<script>` tag, now use _Bundles → Select Bundle Item…_ and switch to “Settings”. Enter `comment` and you should see all comment variables for your current scope. You can initiate an edit from this dialog.

**Comment by infininight** [comment]
Corrected in 7926b09bf72f717fa586d892ea9bb51510c96f33, we now use a surrounding `meta.embedded` scope and only use the the `source.js` scope for the interior. A remaining issue is when you are on the same line as the opening tag but after it's closing `>`. Thanks for the report!
